lib/_http_agent.js

```diff
@@ -131,7 +133,7 @@ function Agent(options) {
     // and could cause unhandled exception.
 
     if (!socket.writable) {
-      socket.destroy();
+      socket.on('error', nop).destroy();
```

Comment:
Is there no other way here? I'm not very happy with these.

Reply:
They are not needed for https://github.com/nodejs/node/issues/36931 as the socket is already destroyed so `'free'` is not emitted as per your fix below.

I would wait for a specific bug before adding them.

Reply:
removed